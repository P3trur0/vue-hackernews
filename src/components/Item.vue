<template>
  <li class="news-item">
    <a :href="item.url">{{ item.title }}</a>
      {{ item.by }} {{ item.url }} {{ item.score }}
  </li>
</template>

<script>
export default {
  name: 'news-item',
  props: ['item']
}
</script>
