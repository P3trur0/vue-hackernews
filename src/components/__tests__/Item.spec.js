import { mount } from 'vue-test-utils'
import Item from '../Item.vue'

describe('Item.vue', () => {
  test('renders item.score', () => {
    const item = {
      score: 10
    }
    const wrapper = mount(Item, {
      propsData: { item } // #A
    })
    expect(wrapper.text()).toContain(item.score) // #B
  })

  test('renders item.by', () => {
    const item = {
      by: 'some author'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.text()).toContain(item.by)
  })

  test('renders item.url', () => {
    const item = {
      url: 'some title'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    expect(wrapper.text()).toContain(item.url)
  })

  test('renders an a tag containing item.title', () => {
    const item = { // #A
      title: 'some title'
    }
    const wrapper = mount(Item, {
      propsData: { item } // #B
    })
    expect(wrapper.find('a').text()).toEqual(item.title) // #C
  })

  test('renders an a tag with href item.url', () => {
    const item = {
      url: 'http://some-url.com'
    }
    const wrapper = mount(Item, {
      propsData: { item }
    })
    const aWrapper = wrapper.find('a')
    expect(aWrapper.attributes().href).toBe(item.url) // #A
  })
})
