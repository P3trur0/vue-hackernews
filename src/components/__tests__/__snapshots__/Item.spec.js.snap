// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item.vue renders correctly 1`] = `
<li class="news-item"><span class="score">1</span> <span class="title"><a href="http://dawn.cs.stanford.edu/" target="_blank" rel="noopener">DAWN: Tools for AI and Data Product Development</a> <span class="host"> (cs.stanford.edu)</span></span>
    <br> <span class="meta"><span class="by">
      by <router-link to="/user/spinwang">spinwang</router-link></span> <span class="comments-link">
      | <router-link to="/item/15969539"> comments</router-link></span> <span class="time">
    1 day ago
  </span></span>
    <!---->
</li>
`;
